tls/extensions/s2n_ec_point_format.c

```diff
@@ -64,7 +64,16 @@ static int s2n_ec_point_format_send(struct s2n_connection *conn, struct s2n_stuf
 
 static int s2n_ec_point_format_recv(struct s2n_connection *conn, struct s2n_stuffer *extension)
 {
-    /**
+    /*
+     *= https://tools.ietf.org/rfc/rfc8422#section-5.1.2
+     *= type=exception
+     *= reason=Incorrect implementations exist in the wild. Skipping validation.
+     *# If the client sends the extension and the extension does not contain
+     *# the uncompressed point format, and the client has used the Supported
+     *# Groups extension to indicate support for any of the curves defined in
+     *# this specification, then the server MUST abort the handshake and
+     *# return an illegal_parameter alert.
+     *
      * Only uncompressed points are supported by the server and the client must include it in
      * the extension. Just skip the extension.
```

Comment:
Same comment as your previous PR-- put the actual code comment before the compliance comment. The comment should be read first.

Reply:
https://github.com/aws/s2n-tls/pull/4202/files#r1329328536

> When a comment comes after the compliance comment I read it as a clarification of the compliance comment.

I think you are referring to this comment? In the case of an `exception`, the comment felt like a clarification to the requirement since our behavior deviates. However, I can change the order.

Reply:
Reordered both the citations: https://github.com/aws/s2n-tls/pull/4202/commits/e201e9a81b6adcaf606d51a8c9b6e27faffdb52d